libevmasm/SemanticInformation.cpp

```diff
@@ -278,12 +282,48 @@ bool SemanticInformation::invalidatesStorage(Instruction _instruction)
 	case Instruction::CREATE:
 	case Instruction::CREATE2:
 	case Instruction::SSTORE:
-		return true;
+		return yul::Effect::Write;
+	case Instruction::EXTCODESIZE:
+	case Instruction::EXTCODEHASH:
+	case Instruction::RETURNDATASIZE:
+	case Instruction::BALANCE:
+	case Instruction::SELFBALANCE:
+	case Instruction::SLOAD:
+		return yul::Effect::Read;
+	// TODO what about keccak256, msize, etc?
+
 	default:
-		return false;
+		return yul::Effect::Pure;
+	}
+}
+
+yul::Effect SemanticInformation::state(Instruction _instruction)
```

Comment:
Maybe we should call this `otherState` which then means, reading or writing anything apart from memory or storage?